pkg/gofr/http_server.go

```diff
@@ -47,15 +50,32 @@ func (s *httpServer) RegisterProfilingRoutes() {
 }
 
 func (s *httpServer) Run(c *container.Container) {
-	var srv *http.Server
+	if s.srv != nil {
+		c.Logf("Server already running on port: %d", s.port)
+		return
+	}
 
 	c.Logf("Starting server on port: %d", s.port)
 
-	srv = &http.Server{
+	s.srv = &http.Server{
 		Addr:              fmt.Sprintf(":%d", s.port),
 		Handler:           s.router,
 		ReadHeaderTimeout: 5 * time.Second,
 	}
 
-	c.Error(srv.ListenAndServe())
+	c.Error(s.srv.ListenAndServe())
+}
+
+func (s *httpServer) Shutdown(ctx context.Context, logger logging.Logger) error {
+	if s.srv == nil {
+		return nil
+	}
+
+	return shutdownWithTimeout(ctx, func(ctx context.Context) error {
+		return s.srv.Shutdown(ctx)
+	}, func() {
+		if err := s.srv.Close(); err != nil {
+			logger.Errorf("Error closing http server: %v", err)
+		}
+	})
```

Comment:
I would have modified shutdownWithTimeout signature to be a func() error for second function

Then handling the error in this function, avoiding to deal with  the logger you have to import here

So something like this
https://github.com/gofr-dev/gofr/pull/837#discussion_r1687479094

Reply:
yups that can also be done, refactored the signature to that